src/simularium/VisGeometry/GeometryStore.ts

```diff
@@ -0,0 +1,400 @@
+        this._geoLoadAttempted = new Map<string, boolean>();
+        this._cachedAssets = new Map<string, string>();
+        this._registry = new Map<string, AgentGeometry>();
+        this.mlogger = jsLogger.get("geometry-store");
+        this._registry.set(DEFAULT_MESH_NAME, {
+            displayType: GeometryDisplayType.SPHERE,
+            geometry: {
+                mesh: new Mesh(VisAgent.sphereGeometry),
+                cancelled: false,
+                instances: new InstancedMesh(
+                    VisAgent.sphereGeometry,
+                    DEFAULT_MESH_NAME,
+                    1
+                ),
+            },
+        });
+        if (loggerLevel) {
+            this.mlogger.setLevel(loggerLevel);
+        }
+    }
+
+    public reset(): void {
+        this._geoLoadAttempted.clear();
+        this._registry.clear();
+        this._registry.set(DEFAULT_MESH_NAME, {
+            displayType: GeometryDisplayType.SPHERE,
+            geometry: {
+                mesh: new Mesh(VisAgent.sphereGeometry),
+                cancelled: false,
+                instances: new InstancedMesh(
+                    VisAgent.sphereGeometry,
+                    DEFAULT_MESH_NAME,
+                    1
+                ),
+            },
+        });
+    }
+
+    public get registry(): Registry {
+        return this._registry;
+    }
+
+    public forEachMesh(iteratee: (geo: MeshGeometry) => void): void {
+        // forEach method for manipulating ThreeJs Mesh objects
+        this._registry.forEach((value) => {
+            const { displayType } = value;
+            if (displayType !== GeometryDisplayType.PDB) {
+                const agentGeo = value as MeshGeometry;
+                iteratee(agentGeo);
+            }
+        });
+    }
+
+    public getGeoForAgentType(entryName: string): AgentGeometry | null {
+        const geo = this._registry.get(entryName);
+        return geo || null;
+    }
+
+    public cancelAll(): void {
+        // note that this leaves cancelled things in the registries.
+        // This should be called before the registries are cleared and probably
+        // only makes sense to do if they are indeed about to be cleared.
+
+        // don't process any queued requests
+        TaskQueue.stopAll();
+        // signal to cancel any pending geometries
+        this._registry.forEach((value) => {
+            value.geometry.cancelled = true;
+        });
+    }
+
+    public cacheLocalAssets(assets: { [key: string]: string }): void {
+        // store local files as data strings until they're ready to be
+        // parsed as geometry.
+        forEach(assets, (value, key) => {
+            const path = checkAndSanitizePath(key);
+            this._cachedAssets.set(path, value);
+        });
+    }
+
+    private createNewSphereGeometry(meshName: string): MeshLoadRequest {
+        /** create new default geometry */
+        return {
+            mesh: new Mesh(VisAgent.sphereGeometry),
```

Comment:
I don't think there's much need to have sphereGeometry live in VisAgent anymore.  either now, or as a TODO, especially when we support CUBE etc